hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -338,6 +340,14 @@ commit ctx st utxo = do
     (_, TxOut ShelleyAddressInEra{} _ _ _) ->
       Right ()
 
+  -- NOTE: Remove this limit once we have more experiments on mainnet.
+  rejectMoreThan100ADA :: NetworkId -> TxOut CtxUTxO -> Either (PostTxError Tx) ()
+  rejectMoreThan100ADA Mainnet output =
+    if selectLovelace (txOutValue output) > 100
+      then Left $ FailedToPostTx "hard-coded limit of 100 ADA for the commit tx on mainnet reached."
```

Comment:
Should: Introduce a dedicated error type, which we can document in the API. Also, should document this in the API :)